speculos.py

```diff
@@ -128,29 +160,40 @@ def run_qemu(s1, s2, app_path, libraries=[], seed=DEFAULT_SEED, debug=False, tra
     args = parser.parse_args()
     args.model.lower()
 
-    if not (args.flushed or args.progressive):
-        if args.model == 'blue':
-            args.progressive = True
+    try:
+        if not (args.flushed or args.progressive):
+            if args.model == 'blue':
+                args.progressive = True
+            else:
+                args.flushed = True
+
+        if args.model != 'blue' and (args.rampage != None or args.pagesize != None):
+            # Extra RAM page only allowed on Blue device
+            raise ValueError(f"Extra RAM page arguments -r (--rampage) & -q (--pagesize)) require '-m blue'")
+
+        if args.text:
+            if args.model != 'nanos':
+                raise ValueError(f"Unsupported model '{args.model}' with -x")
+            if args.ontop:
```

Comment:
`parser.add_mutually_exclusive_group()` maybe?

Reply:
Won't change: 
Actually argument `--ontop` is already in a mutually exclusive group with `--headless`. And based on its description, the `--text` argument overrides `--headless`. So it cannot be added to the same group. And I don't think adding a 2nd mutualy exclusive group with `--ontop` and `--text` would work since `--ontop ` is already in another group.